loopy/transform/merge_arrays.py

```diff
@@ -0,0 +1,107 @@
+__copyright__ = "Copyright (C) 2022 Isuru Fernando"
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+__doc__ = """
+.. currentmodule:: loopy
+.. autofunction:: merge_temporary_arrays
+"""
+
+from loopy.kernel.data import auto
+from loopy.symbolic import SubstitutionRuleMappingContext
+from loopy.kernel import LoopKernel
+from loopy.translation_unit import for_each_kernel
+
+import pymbolic.primitives as prim
+from pytools import all_equal
+
+
+@for_each_kernel
+def merge_temporary_arrays(kernel, array_names, new_name=None, axis_nr=0):
```

Comment:
In keeping with numpy terminology, would "concatenate" make sense as a name and/or as far as semantics go?

https://numpy.org/doc/stable/reference/generated/numpy.concatenate.html

Reply:
`concatenate` for me means take two arrays and return a new array concatenated. Here we do a concatenation, but we replace the uses of each array with the concatenated one. Not sure `concatenate` captures that behaviour.

Reply:
Maybe `concatenated_memory_layout_for_temporaries`?

cc @kaushikcfd

Reply:
> concatenate for me means take two arrays and return a new array concatenated

Makes sense. Maybe "concatenate_memory_layout_of_temporaries"?

Reply:
Done in 84204af

Reply:
I renamed this `concatenate_arrays`, since the memory layout is not technically visible at the level at which the transformation operates.